### PR TITLE
tiflash: Add "marker" for security.redact_info_log in tiflash.toml

### PR DESCRIPTION
**Pull request by Oreoxmt**
### What is changed, added or deleted? (Required)

Enhance log densensitation in tiflash.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/17627
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tiflash/tiflash-configuration.md [hunk 1]
```diff
@@ -253,8 +253,10 @@ delta_index_cache_size = 0
 
 ## Security settings take effect starting from v4.0.5.
 [security]
-    ## New in v5.0. This configuration item enables or disables log redaction. If the configuration value
-    ## is set to true, all user data in the log will be replaced by ?.
+    ## New in v5.0. This configuration item enables or disables log redaction. Value options: true, false, marker. The marker option is introduced in v8.2.0
```
Comment (qiancai) [inline tiflash/tiflash-configuration.md:256]:
```suggestion
    ## New in v5.0. This configuration item enables or disables log redaction. Value options: true, false, marker. The marker option is introduced in v8.2.0.
```

File: tiflash/tiflash-configuration.md [hunk 2]
```diff
@@ -253,8 +253,10 @@ delta_index_cache_size = 0
...
+    ## The default value is false, which means that log redaction is disabled.
+    ## If the configuration item is set to true, all user data in the log is replaced by `?`.
+    ## If the configuration item is set to marker, all user data in the log is wrapped in `‹ ›`. If the input has `‹`, it is escaped as `‹‹`, and `›` is escaped as `››`. Based on the marked logs, you can decide whether to desensitize the marked information when the logs are displayed.
```
Comment (qiancai) [inline tiflash/tiflash-configuration.md:259]:
```suggestion
    ## If the configuration item is set to marker, all user data in the log is wrapped in `‹ ›`. If user data contains `‹` or `›`, `‹` is escaped as `‹‹`, and `›` is escaped as `››`. Based on the marked logs, you can decide whether to desensitize the marked information when the logs are displayed.
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

File: tiflash/tiflash-configuration.md (hunk 2 above)
Comment (qiancai) [reply tiflash/tiflash-configuration.md:259]: see https://github.com/pingcap/docs-cn/pull/17627#discussion_r1642424899

Comment (JaySon-Huang) [reply tiflash/tiflash-configuration.md:259]: maybe we should also change the following text "If the input has `‹` ... " here

https://github.com/pingcap/docs/blob/cd719695c040a41c1463f3fdba42f80aed8803d3/system-variables.md?plain=1#L4936

**Review by JaySon-Huang — APPROVED** [review]

**Pull request merged**